src/Components/EntityState.cs

```diff
@@ -0,0 +1,8 @@
+﻿namespace Components
+{
+	public interface EntityState
+	{
+		bool TryAdd<Component>(Component component);
+		Component Get<Component>();
```

Comment:
@danielcirket null vs exception while valid, is not the point i was aiming for. Its more to do with describing the API so that the sematics are clearer for the consumer. So:

    EntityState state;
    if (entityState.TryGet<MyComponent>(out state))
        // do stuff 

vs

    var state = entityState.Get<MyComponent>();
    if(state != null)
         // do stuff

In the latter, if the implementation of `entityState` changes from returning a null to returning an exception then our code is no longer valid because we are binding our code to a known implementation of `EntityState`. However, in the former, the API is guaranteeing the method call will be consistent (as in - returning a bool, setting the out param).